pkg/genesis/vesting_accounts.go

```diff
@@ -0,0 +1,158 @@
+) (account authtypes.GenesisAccount, balance banktypes.Balance, err error) {
+	sdkAddr, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
+		return account, balance, err
+	}
+
+	balance = banktypes.Balance{
+		Address: address,
+		Coins:   initUnlockedCoins.Add(vestingBalance...),
+	}
+
+	bAccount := authtypes.NewBaseAccountWithAddress(sdkAddr)
+	vAccount := vestingtypes.NewDelayedVestingAccount(bAccount, vestingBalance, endTime.Unix())
+
+	return authtypes.GenesisAccount(vAccount), balance, nil
+}
+
+func NewGenesisPeriodicVestingAccount(
+	address string,
+	vestingBalance,
+	initUnlockedCoins sdk.Coins,
+	startTime time.Time,
+	periods []vestingtypes.Period,
+) (account authtypes.GenesisAccount, balance banktypes.Balance, err error) {
+	sdkAddr, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
+		return account, balance, err
+	}
+
+	balance = banktypes.Balance{
+		Address: address,
+		Coins:   initUnlockedCoins.Add(vestingBalance...),
+	}
+
+	bAccount := authtypes.NewBaseAccountWithAddress(sdkAddr)
+	vAccount := vestingtypes.NewPeriodicVestingAccount(bAccount, vestingBalance, startTime.Unix(), periods)
+
+	return authtypes.GenesisAccount(vAccount), balance, nil
+}
+
+func NewGenesisContinuousVestingAccount(
+	address string,
+	vestingBalance,
+	initUnlockedCoins sdk.Coins,
+	startTime, endTime time.Time,
+) (account authtypes.GenesisAccount, balance banktypes.Balance, err error) {
+	sdkAddr, err := sdk.AccAddressFromBech32(address)
+	if err != nil {
+		return account, balance, err
+	}
+
+	balance = banktypes.Balance{
+		Address: address,
+		Coins:   initUnlockedCoins.Add(vestingBalance...),
+	}
+
+	bAccount := authtypes.NewBaseAccountWithAddress(sdkAddr)
+	vAccount := vestingtypes.NewContinuousVestingAccount(bAccount, vestingBalance, startTime.Unix(), endTime.Unix())
+
+	return authtypes.GenesisAccount(vAccount), balance, nil
+}
+
+// AddAccountsToGenesisState adds the provided accounts to the genesis state (gs) map for the auth module.
+// It takes the raw genesis state (gs) and a variadic number of GenesisAccount objects (accounts) as inputs.
+// Then, it updates the given genesis state and returns it.
+func AddAccountsToGenesisState(encCfg encoding.Config, gs map[string]json.RawMessage, accounts ...authtypes.GenesisAccount) (map[string]json.RawMessage, error) {
+	var authGenState authtypes.GenesisState
+	err := encCfg.Codec.UnmarshalJSON(gs[authtypes.ModuleName], &authGenState)
+	if err != nil {
+		return gs, err
+	}
+
+	pAccs, err := authtypes.PackAccounts(accounts)
+	if err != nil {
+		return gs, err
+	}
+
+	// set the accounts in the genesis state
+	authGenState.Accounts = append(authGenState.Accounts, pAccs...)
+	gs[authtypes.ModuleName] = encCfg.Codec.MustMarshalJSON(&authGenState)
+
+	return gs, nil
+}
+
+// AddBalancesToGenesisState adds the provided balances to the genesis state (gs) for the bank module.
+// It takes the raw genesis state (gs) and a variadic number of Balance objects (balances) as inputs.
```

Comment:
> and a variadic number of Balance objects (balances)

this doesn't appear to be the case, if not then we should update these docs

Reply:
yeah it seems like the comment is outdated